src/vs/workbench/contrib/terminal/browser/links/terminalLinkProviderAdapter.ts

```diff
@@ -0,0 +1,97 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Emitter } from 'vs/base/common/event';
+import { Disposable } from 'vs/base/common/lifecycle';
+import { localize } from 'vs/nls';
+import { IInstantiationService } from 'vs/platform/instantiation/common/instantiation';
+import { ITerminalSimpleLink, ITerminalLinkDetector, TerminalLinkType } from 'vs/workbench/contrib/terminal/browser/links/links';
+import { TerminalLink } from 'vs/workbench/contrib/terminal/browser/links/terminalLink';
+import { XtermLinkMatcherHandler } from 'vs/workbench/contrib/terminal/browser/links/terminalLinkManager';
+import { IBufferLine, ILink, ILinkProvider } from 'xterm';
+
+/**
+ * Wrap a link detector object so it can be used in xterm.js
+ */
+export class TerminalLinkProviderAdapter extends Disposable implements ILinkProvider {
```

Comment:
maybe should be named `TerminalLinkDetectorAdapter` or `TerminalLinkDetectorWrapper`?

Reply:
Yeah the first one's probably better